### PR TITLE
Ts typing 

### PR DESCRIPTION
**Pull request by martinRenou**
Using a type check so we don't have to cast with `as ...`

### CODE REVIEW

File: js/src/Axis.ts [hunk 1]
```diff
@@ -112,9 +118,9 @@ export class Axis extends WidgetView {
         } else if (num_ticks !== undefined && num_ticks !== null) {
             this.axis.tickValues(this.get_ticks_from_array_or_length());
         } else {
-            if (this.axis_scale.model.type === "ordinal") {
+            if (this.axis_scale instanceof OrdinalScale) {
```
Comment (davidbrochart) [inline js/src/Axis.ts:121]: Is this going to work? `axis_scale` is still of type `Scale`:
https://github.com/davidbrochart/bqplot/blob/5c9ad992431b8251b79f1210c597ff245508d56c/js/src/Axis.ts#L925

File: js/src/Axis.ts [hunk 2]
```diff
@@ -112,9 +118,9 @@ export class Axis extends WidgetView {
...
                 this.axis.tickValues(this.axis_scale.scale.domain());
-            } else if (this.axis_scale.model.type === "date") {
+            } else if (this.axis_scale instanceof DateScale) {
```
Comment (davidbrochart) [inline js/src/Axis.ts:123]: Should we get rid of the `type` attribute in each scale, if we now rely on the instance type?
https://github.com/davidbrochart/bqplot/blob/e4716a4e7a13f08a342359591a9a764aa7710543/js/src/DateScaleModel.ts#L40

File: js/src/Axis.ts [hunk 3]
```diff
@@ -153,16 +159,13 @@ export class Axis extends WidgetView {
                     }
                     this.axis.tickValues(useticks);
                 }
-            } else {
-                const scale = this.axis_scale.scale as (d3.ScaleLinear<number, number> |
-                                                        d3.ScaleTime<Date, number> |
-                                                        d3.ScaleLogarithmic<number, number>);
-                this.axis.tickValues(scale.ticks());
+            } else if (this.axis_scale instanceof LinearScale) {
```
Comment (davidbrochart) [inline js/src/Axis.ts]: This branch doesn't include date scale or log scale?

File: js/src/Axis.ts (hunk 1 above)
Comment (martinRenou) [reply js/src/Axis.ts:121]: Yes. An `OrdinalScale` is a `Scale` :)

File: js/src/Axis.ts (hunk 2 above)
Comment (martinRenou) [reply js/src/Axis.ts:123]: I think so yes. Although some bqplot plugins might still rely on them, so maybe we should show a deprecation warning for now.

File: js/src/Axis.ts (hunk 3 above)
Comment (martinRenou) [reply js/src/Axis.ts]: I added DateScale, thanks!

LogScale already has its own branch in this method.

File: js/src/Axis.ts (hunk 1 above)
Comment (davidbrochart) [reply js/src/Axis.ts:121]: But is it possible to identify the type of `axis_scale`, if it's just typed as a `Scale`? Wouldn't all of these checks always be true?

Comment (martinRenou) [reply js/src/Axis.ts:121]: Those checks are done at runtime, it compiles to this JavaScript code: 
```javascript
if (this.axis_scale instanceof OrdinalScale_1.OrdinalScale) {
    this.axis.tickValues(this.axis_scale.scale.domain());
}
```

The information that `axis_scale` is a `Scale` does not exist in the JavaScript code.

**Pull request merged**